.github/workflows/wheels.yml

```diff
@@ -124,6 +127,13 @@ jobs:
         with:
           python-version: '3.9'  # update once build dependencies are available
 
+      - name: Set pip extra options
+        if: github.event_name == 'schedule'
+        run: |
+          # Test with nightly dependencies for nightly builds
+          echo "PIP_EXTRA_INDEX=https://pypi.anaconda.org/scipy-wheels-nightly/simple" >> $GITHUB_ENV
+          echo "PIP_PRE=1" >> $GITHUB_ENV
```

Comment:
As discussed in https://github.com/scikit-learn/scikit-learn/pull/24446/files#r1000353644, let's focus this PR on building and publishing 3.11 wheels against stable dependencies and remove the use of dev dependencies from this PR.

We can tackle nightly testing against dev dependencies (without uploading) in a dedicated PR.

Do do that, we could introduce a second github actions scheduled event with a specific env variable/config flag to also disable the final upload to anaconda or pypi.

We will also have to re-implement the `[scipy-dev]` commit message trigger currently implemented in our Azure Pipelines config and delete that config from Azure Pipelines.

```suggestion
```

Reply:
@cmarmo I did not have time to write that comment and you addressed it concurrently :)